### PR TITLE
[Program Card Images] Show a preview of the program card in the admin UI

### PR DESCRIPTION
**Pull request by caitlinshk**
### Description

This PR updates the admin UI for uploading a program image to also include a preview of the program card with the uploaded image. This should help admins understand how their image will get cropped when it's shown to applicants.

Most of this PR is refactoring the code that renders a program card out of `ProgramIndexView` and into `ProgramCardViewRenderer` so that it can be reused in the admin UI.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible) -- **existing browser tests should be sufficient, see updated screenshots**
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### Admin UI
1. Edit your `/etc/hosts` file to also contain this line:
```
127.0.0.1 civiform-local-s3-public.s3.localhost.localstack.cloud
```

2. Enable the program card images flag.
3. As an admin, edit a program and then click the "Edit program image" button.
4. On that page, verify that the right side of the page has a "What the applicant will see" header and a program card representing the program in its current form.
5. Upload a program image.
7. Click "Save image".
8. Verify the page reloads and that the program card on the right side now displays the image.

#### Applicant UI

Open the applicant homepage and verify that the program cards still render correctly.

### Issue(s) this completes

Epic: https://github.com/civiform/civiform/issues/5676

Specific task: https://github.com/civiform/civiform/issues/5999

### CODE REVIEW

**Comment by caitlinshk** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/6227?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #6041** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6041?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6042** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6042?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #6049** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6049?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #6050** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6050?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #6200** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6200?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #6218** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6218?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #6227** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6227?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #6043** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6043?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6227?utm_source=stack-comment).

File: server/app/views/applicant/ProgramCardViewRenderer.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+import views.components.Icons;
+import views.components.LinkElement;
+import views.components.Modal;
+import views.components.TextFormatter;
+import views.style.ApplicantStyles;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+
+/** A renderer to create an individual program information card. */
+public final class ProgramCardViewRenderer {
```
Comment (melanie-exygy) [inline server/app/views/applicant/ProgramCardViewRenderer.java:61]: praise:  Thanks for making this reusable!  I also have a feeling that splitting this out could help us when we migrate to a templating tool and split the markup into potentially "hot-swappable" chunks.

File: server/app/views/admin/programs/ProgramImageView.java [hunk 2]
```diff
@@ -172,12 +189,45 @@ private StorageUploadRequest createStorageUploadRequest(ProgramDefinition progra
...
+  private DivTag renderCurrentProgramCard(Http.Request request, ProgramDefinition program) {
+    DivTag currentProgramCardSection =
+        div().with(h2("What the applicant will see").withClasses("mb-4"));
+
+    Optional<CiviFormProfile> profile = profileUtils.currentUserProfile(request);
+    Long applicantId;
+    try {
+      applicantId = profile.get().getApplicant().get().id;
+    } catch (NoSuchElementException | ExecutionException | InterruptedException e) {
+      return currentProgramCardSection.with(p("Applicant preview cannot be rendered"));
```
Comment (melanie-exygy) [inline server/app/views/admin/programs/ProgramImageView.java:201]: suggestion (non-blocking):  I wonder if we should also log a warning here for troubleshooting with a more specific message like "Could not fetch applicant id".

File: browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png
[the remark this answers is not in the document]
Comment (caitlinshk) [reply browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png:1]: Ugh, I think my PR stacking got in your way. My PRs were stacked with 6200 on bottom, then 6218, then 6227 (this one). PRs 6200 and 6218 got submitted before your review, so I think this PR was no longer had PRs 6200 and 6218 as a diffbase. Instead, it seems like this PR is showing the changes from all three PRs (6200, 6218, *and* itself).

I just synced this branch and pushed and the changes from the old PRs are no longer showing.

File: server/app/views/admin/programs/ProgramImageView.java (hunk 2 above)
Comment (caitlinshk) [reply server/app/views/admin/programs/ProgramImageView.java]: Agreed, done!

File: browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png
[the remark this answers is not in the document]
Comment (caitlinshk) [reply browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png:1]: And, apologies for you having to look through way more files than you should've! I didn't realize that's how the stacking worked.

File: browser-test/src/admin_program_image.test.ts [hunk 3]
```diff
@@ -126,10 +126,35 @@ describe('Admin can manage program image', () => {
       await adminPrograms.goToProgramImagePage(programName)
     })
 
+    it('form is correctly formatted', async () => {
```
[the remark this answers is not in the document]
Comment (caitlinshk) [reply browser-test/src/admin_program_image.test.ts]: (See https://github.com/civiform/civiform/pull/6227#discussion_r1433349853, I think it's the same issue)

File: browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png
[the remark this answers is not in the document]
Comment (melanie-exygy) [reply browser-test/image_snapshots/admin_program_visibility_test/program-visibility-ti-only-visible-to-ti.png:1]: I figured it must be something like that.  Thanks for sorting it all out.

**Review by melanie-exygy — APPROVED** [review]

**Comment by caitlinshk** [comment]
### Merge activity

* **Dec 21, 5:15 PM**: @@caitlinshk merged this pull request with [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6227).

**Pull request merged**